hr_auto_manager_group_membership/models/hr_employee.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2022 Coop IT Easy SCRLfs
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+MANAGER_GROUP = "hr_auto_manager_group_membership.group_employee_manager"
+MANAGER_GROUP_DOMAIN = [
+    ("model", "=", "res.groups"),
+    ("module", "=", "hr_auto_manager_group_membership"),
+    ("name", "=", "group_employee_manager"),
+]
+
+
+class HrEmployee(models.Model):
+
+    _inherit = "hr.employee"
+
+    @api.model
+    def create(self, vals):
+        result = super().create(vals)
+        result._update_manager_group_membership()
+        if result.parent_id:
+            result.parent_id._update_manager_group_membership()
+        return result
+
+    @api.multi
+    def write(self, vals):
+        parent_ids = [rec.parent_id for rec in self]
+        result = super().write(vals)
+        for rec, parent_id in zip(self, parent_ids):
+            rec._update_manager_group_membership()
+            if parent_id and parent_id != rec.parent_id:
+                parent_id._update_manager_group_membership()
```

Comment:
This needs a comment. I can't divine under which circumstances this would be true.

Reply:
comment added.